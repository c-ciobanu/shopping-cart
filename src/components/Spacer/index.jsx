import React from "react";
import PropTypes from "prop-types";

import { StyledDiv } from "./styled";
import { sizes } from "./consts";

const sizeNames = Object.keys(sizes);

function Spacer({ size }) {
	return <StyledDiv height={sizes[size]} />;
}

Spacer.propTypes = {
	size: PropTypes.oneOf(sizeNames).isRequired
};

export default Spacer;
