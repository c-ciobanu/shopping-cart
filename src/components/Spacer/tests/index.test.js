import React from "react";
import { render } from "@testing-library/react";

import Spacer from "../index";

describe("<Spacer />", () => {
	it("Expect to not log errors in console", () => {
		const spy = jest.spyOn(global.console, "error");
		render(<Spacer size="xxl" />);
		expect(spy).not.toHaveBeenCalled();
	});

	it("Should render and match the snapshot", () => {
		const {
			container: { firstChild }
		} = render(<Spacer size="xxl" />);
		expect(firstChild).toMatchSnapshot();
	});
});
