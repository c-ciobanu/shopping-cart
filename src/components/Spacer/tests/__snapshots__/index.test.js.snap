// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Spacer /> Should render and match the snapshot 1`] = `
<div
  class="sc-AykKC kZrTuu"
  height="40px"
/>
`;
